frontend/src/core/comments/components/AddComment.js

```diff
@@ -25,69 +41,207 @@ export default function AddComments(props: Props) {
+                    case 'Enter':
+                        event.preventDefault();
+                        Transforms.select(editor, target);
+                        insertMention(editor, chars[index]);
+                        disableHistoryScroll(false);
+                        setTarget(null);
+                        break;
+                    case 'Escape':
+                        event.preventDefault();
+                        disableHistoryScroll(false);
+                        setTarget(null);
+                        break;
+                    default:
+                        return;
+                }
+            }
+        },
+        [chars, editor, index, target]
+    );
+
+    const onKeyDown = (event: SyntheticKeyboardEvent<>) => {
+        if (event.key === 'Enter' && !event.shiftKey) {
+            event.preventDefault();
+            submitComment();
+        }
+    }
+    
     if (!user) {
         return null;
     }
 
-    const handleOnChange = () => {
-        const textAreaLineHeight = 24;
-        commentInput.current.rows = minRows;
+    const onChange = (value) => {
+        setValue(value);
+        const { selection } = editor;
 
-        const currentRows = Math.trunc(commentInput.current.scrollHeight / textAreaLineHeight);
-
-        if (currentRows < maxRows) {
-            commentInput.current.rows = currentRows;
+        if (selection && Range.isCollapsed(selection)) {
+            const [start] = Range.edges(selection);
+            const wordBefore = Editor.before(editor, start, { unit: 'word' });
+            const before = wordBefore && Editor.before(editor, wordBefore);
+            const beforeRange = before && Editor.range(editor, before, start);
+            const beforeText = beforeRange && Editor.string(editor, beforeRange);
+            const beforeMatch = beforeText && beforeText.match(/^@(\w+)$/);
+            const after = Editor.after(editor, start);
+            const afterRange = Editor.range(editor, start, after);
+            const afterText = Editor.string(editor, afterRange);
+            const afterMatch = afterText.match(/^(\s|$)/);
+    
+            if (beforeMatch && afterMatch) {
+                setTarget(beforeRange);
+                setSearch(beforeMatch[1]);
+                setIndex(0);
+                return;
+            }
         }
-        else {
-            commentInput.current.rows = maxRows;
+    
+        setTarget(null);
+    }
+
+    const Portal = ({ children }) => {
+        if (!document.body) {
+            return null;
         }
+        return ReactDOM.createPortal(children, document.body);
     }
 
-    const handleOnKeyDown = (event: SyntheticKeyboardEvent<>) => {
-        if (event.keyCode === 13 && event.shiftKey === false) {
-            submitComment(event);
+    const disableHistoryScroll = (mentionsActive: boolean) => {
+        const element = document.querySelector(
+            '.history .mentions-disable-scroll'
+        );
```

Comment:
This only accounts for the current scroll behavior within the `History` panel. 

I was going to set it up with a `ref` on the `history ul` but that caused conflicts with `Team Comments` and before I went too far with that I wanted to be sure this was even a viable solution.